spring-vault-core/src/main/java/org/springframework/vault/authentication/IpAddressUserId.java

```diff
@@ -34,7 +36,7 @@ public String createUserId() {
 			return Sha256.toSha256(InetAddress.getLocalHost().getHostAddress());
 		}
 		catch (IOException e) {
-			throw new IllegalStateException(e);
+			throw new VaultClientException("While creating userId", e);
```

Comment:
This could be a `VaultException`.

Reply:
Yeah I was reserving VaultException as a catchall container root. This (arguably) is a VaultClientException because it occurs when on on the client side, something not to do with the specific call to Vault occurred.

Reply:
Throwing/catching `VaultException` would typically result in just logging or application shutdown (in an extreme case). There's usually nothing you can do from an application perspective to recover from a `VaultException`.